ci-operator/step-registry/optional-operators/cvp-common/scorecard/optional-operators-cvp-common-scorecard-commands.sh

```diff
@@ -5,47 +5,64 @@ set -o errexit
 set -o pipefail
 set -x
 
+# Runs the supplied command until the non-empty output file is created
+# or retries_max is reached.
+# Takes 2 arguments - the command/function and the output file
 run_scorecard() {
-        local retries_max=3
-        local attempt_num=1
-        until [[ $(jq . "$2") ]]
-        
-        do
-            "$1"
-            if (( attempt_num==retries_max ))
-            then
-                    echo "Retry attempt number: $attempt_num of $retries_max of scorecard tests failed. No more attempts. "
-                    return 1
-            else
-                    echo "Retry attempt number: $attempt_num of $retries_max failed. Retrying."
-                    sleep 5
-                    ((attempt_num++))
-            fi
-        done
+  local retries_max=3
+  local attempt_num=1
+  until [ -s "$2" ]
+
+  do
+    if (( attempt_num>retries_max )); then
+      echo "All $retries_max attempts of scorecard tests failed. No more attempts. "
+      return 1
+    else
+      echo "Attempt $attempt_num of $retries_max."
+      sleep 5
+      ((attempt_num++))
+    fi
+      "$1" "$2"
+  done
 }
 
+# Runs the basic scorecard tests using the prepared scorecard config
+# Takes 1 argument as the output file for the scorecard command
 basic_tests() {
-operator-sdk scorecard --config "${SCORECARD_CONFIG}" \
+  local OUTPUT_FILE=$1
+  operator-sdk scorecard --config "${SCORECARD_CONFIG}" \
                        --namespace "${NAMESPACE}" \
                        --kubeconfig "${KUBECONFIG}" \
                        --verbose \
                        --output json \
-                       "${OPERATOR_DIR}" > "${ARTIFACT_DIR}"/scorecard-output-basic.json || true
+                       "${OPERATOR_DIR}" > "${OUTPUT_FILE}" || true
 }
 
+# Runs the custom scorecard using the config in the operator bundle image
+# Takes 1 argument as the output file for the scorecard command
+# If the CUSTOM_SCORECARD_TESTCASE is set, runs in single test case mode:
+#    Outputs the results in xunit format and stores them in the ARTIFACT_DIR
+#    Uses the selector option to execute just a single test from the config
 custom_tests() {
-operator-sdk scorecard \
-    --namespace="${NAMESPACE}" \
-    --kubeconfig "${KUBECONFIG}" \
-    --verbose \
-    --output json \
-    --wait-time 3000s \
-    --service-account "${SCORECARD_SERVICE_ACCOUNT}" \
-    "${OPERATOR_DIR}" > "${ARTIFACT_DIR}"/scorecard-output-custom.json || true
+  local OUTPUT_FILE=$1
+  ADDITIONAL_OPTIONS=""
+  if [[ -n "${CUSTOM_SCORECARD_TESTCASE}" && "${TEST_MODE}" == "msp" ]]; then
```

Comment:
Removing ` && "${TEST_MODE}" == "msp"` will simplify testing.
Cause we would be able to test `CUSTOM_SCORECARD_TESTCASE` even with cvp test mode.
But it will not be used in production.

cvp will pass `CUSTOM_SCORECARD_TESTCASE` in in case of `msp` test mode